### PR TITLE
docs: use server relative URL

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Change URL from a relative link to a server relative link

## Context:

Steps to reproduce the current problem:

1. Get bot message: "Hey you need to fill out the new package manager questionnaire".
2. User copy/pastes questionnaire into new comment on their issue.
3. Navigate to comment on issue.
4. If you click on the link in "I've read the adding a package manager documentation." you get a "file not found" warning from GitHub.

I think it makes more sense to use a server relative link, as people will click on the link online more often than they will use the link locally on a cloned repository.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/new-package-manager-template.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 **Did you read our documentation on adding a package manager?**
 
-- [ ] I've read the [adding a package manager](../../docs/development/adding-a-package-manager.md) documentation.
+- [ ] I've read the [adding a package manager](https://github.com/renovatebot/renovate/blob/main/docs/development/adding-a-package-manager.md) documentation.
```
Comment (viceice) [inline docs/development/new-package-manager-template.md:5]:
```suggestion
- [ ] I've read the [adding a package manager](/renovatebot/renovate/blob/HEAD/docs/development/adding-a-package-manager.md) documentation.
```

What about server relative url?

**Comment by HonkingGoose** [comment]
Are you sure that will work properly once deployed? It seems broken from the GitHub PR file preview "rich diff"?

Current "server relative link" from GitHub "rich diff" preview:

```
https://github.com/renovatebot/renovate/blob/main/renovatebot/renovate/blob/HEAD/docs/development/adding-a-package-manager.md
```

Working link to file on `HEAD`:

```
https://github.com/renovatebot/renovate/blob/HEAD/docs/development/new-package-manager-template.md
```

**Comment by viceice** [comment]
> Are you sure that will work properly once deployed? It seems broken from the GitHub PR file preview "rich diff"?
> 
> Current "server relative link" from GitHub "rich diff" preview:
> 
> ```
> https://github.com/renovatebot/renovate/blob/main/renovatebot/renovate/blob/HEAD/docs/development/adding-a-package-manager.md
> ```
> 
> Working link to file on `HEAD`:
> 
> ```
> https://github.com/renovatebot/renovate/blob/HEAD/docs/development/new-package-manager-template.md
> ```

🤔 Not 💯 sure, but it seems a GitHub UI bug. Those links work when used in issues.

**Comment by HonkingGoose** [comment]
> 🤔 Not 💯 sure, but it seems a GitHub UI bug. Those links work when used in issues.

It works for me too when I tried it in a issue preview screen. 😉

**Comment by HonkingGoose** [comment]
This is ready for a final review now, sorry for the mess/noise I caused with my force pushing, I wanted to clean up the final commit message. 🙈

**Review by viceice — APPROVED** [review]

**Pull request merged**